core/arch/arm/mm/tee_mmu.c

```diff
@@ -629,21 +629,27 @@ void teecore_init_ta_ram(void)
 {
 	vaddr_t s;
 	vaddr_t e;
+	paddr_t ps;
+	paddr_t pe;
 
 	/* get virtual addr/size of RAM where TA are loaded/executedNSec
 	 * shared mem allcated from teecore */
 	core_mmu_get_mem_by_type(MEM_AREA_TA_RAM, &s, &e);
+	ps = virt_to_phys((void *)s);
+	TEE_ASSERT(ps);
+	pe = virt_to_phys((void *)(e - 1)) + 1;
+	TEE_ASSERT(pe);
 
-	TEE_ASSERT((s & (CORE_MMU_USER_CODE_SIZE - 1)) == 0);
-	TEE_ASSERT((e & (CORE_MMU_USER_CODE_SIZE - 1)) == 0);
+	TEE_ASSERT((ps & (CORE_MMU_USER_CODE_SIZE - 1)) == 0);
+	TEE_ASSERT((pe & (CORE_MMU_USER_CODE_SIZE - 1)) == 0);
```

Comment:
They are checking the the start and end are properly aligned. Not really needed in my opinion, but I was fixing the va and pa mixup in this patch.